packages/SwingSet/src/kernel/state/kernelKeeper.js

```diff
@@ -553,14 +557,16 @@ export default function makeKernelKeeper(kernelStorage, kernelSlog) {
    * Allocate a new koid.
    *
    * @param {string} ownerID
-   * @param {undefined | bigint} id
+   * @param {bigint} [id]
    * @returns {string}
    */
-  function addKernelObject(ownerID, id = undefined) {
+  function addKernelObject(ownerID, id) {
```

Comment:
```suggestion
  function addKernelObject(ownerID, id = undefined) {
```
Please don't remove the JS-level indicator that a parameter is optional. Ideally, we'd have a lint rule ensuring that the TS-level and JS-level indicators are consistent with each other. Until then, we should strive to maintain consistency manually and by catching it in review.

Reply:
I switched this to use JSDoc, not TypeScript. https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

Does JSDoc not serve as a JS-level indicator?

As for TypeScript, `= undefined` is a deceptive hint in the absence of additional type annotation.

Reply:
> Does JSDoc not serve as a JS-level indicator?

I was referring to these as TS-level indicators. As far as the JS language is concerned, there is no JSDoc or TS-in-JSDoc. These are all just comments. To be explicit

...  Ideally, we'd have a lint rule ensuring that the TS-in-JSDoc-comment-level and JS-code-level indicators are consistent with each other ...

Another reason is that the TS-in-JSDoc-comments are often textually distant from the JS-code. Even though `= undefined` does not actually change the formal semantics of the JS-code, it changes the meaning of the code as understood by readers.

Reply:
I still disagree but I've reverted this change as it's not necessary for the PR.

Reply:
Thanks. Resolving.